staging/src/k8s.io/kubectl/pkg/cmd/create/create_pdb.go

```diff
@@ -60,54 +95,162 @@ func NewCmdCreatePodDisruptionBudget(f cmdutil.Factory, ioStreams genericcliopti
+	if err != nil {
+		return err
+	}
+	o.Client, err = policyclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	o.CreateAnnotation = cmdutil.GetFlagBool(cmd, cmdutil.ApplyAnnotationsFlag)
+
+	o.DryRunStrategy, err = cmdutil.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+	dynamicClient, err := f.DynamicClient()
+	if err != nil {
+		return err
+	}
+	discoveryClient, err := f.ToDiscoveryClient()
+	if err != nil {
+		return err
+	}
+	o.DryRunVerifier = resourcecli.NewDryRunVerifier(dynamicClient, discoveryClient)
+
+	o.Namespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.PrintObj = func(obj runtime.Object) error {
+		return printer.PrintObj(obj, o.Out)
 	}
 
-	return o.CreateSubcommandOptions.Complete(f, cmd, args, generator)
+	return nil
+}
+
+// Validate checks to the PodDisruptionBudgetOpts to see if there is sufficient information run the command
+func (o *PodDisruptionBudgetOpts) Validate() error {
+	if len(o.Name) == 0 {
+		return fmt.Errorf("name must be specified")
+	}
+
+	if len(o.Selector) == 0 {
+		return fmt.Errorf("a selector must be specified")
+	}
+
+	if len(o.MaxUnavailable) == 0 && len(o.MinAvailable) == 0 {
+		return fmt.Errorf("one of min-available or max-unavailable must be specified")
+	}
+
+	if len(o.MaxUnavailable) > 0 && len(o.MinAvailable) > 0 {
+		return fmt.Errorf("min-available and max-unavailable cannot be both specified")
+	}
+
+	return nil
 }
 
 // Run calls the CreateSubcommandOptions.Run in PodDisruptionBudgetOpts instance
 func (o *PodDisruptionBudgetOpts) Run() error {
-	return o.CreateSubcommandOptions.Run()
+	podDisruptionBudget, err := o.createPodDisruptionBudgets()
+	if err != nil {
+		return err
+	}
+
+	if err := util.CreateOrUpdateAnnotation(o.CreateAnnotation, podDisruptionBudget, scheme.DefaultJSONEncoder()); err != nil {
+		return err
+	}
+
+	if o.DryRunStrategy != cmdutil.DryRunClient {
+		createOptions := metav1.CreateOptions{}
+		if o.FieldManager != "" {
+			createOptions.FieldManager = o.FieldManager
+		}
+		if o.DryRunStrategy == cmdutil.DryRunServer {
+			if err := o.DryRunVerifier.HasSupport(podDisruptionBudget.GroupVersionKind()); err != nil {
+				return err
+			}
+			createOptions.DryRun = []string{metav1.DryRunAll}
+		}
+		podDisruptionBudget, err = o.Client.PodDisruptionBudgets(o.Namespace).Create(context.TODO(), podDisruptionBudget, createOptions)
+		if err != nil {
+			return fmt.Errorf("failed to create poddisruptionbudgets: %v", err)
+		}
+	}
+	return o.PrintObj(podDisruptionBudget)
+}
+
+func (o *PodDisruptionBudgetOpts) createPodDisruptionBudgets() (*policyv1beta1.PodDisruptionBudget, error) {
+	namespace := ""
+	if o.EnforceNamespace {
+		namespace = o.Namespace
+	}
+
+	fmt.Println(policyv1beta1.SchemeGroupVersion.String())
```

Comment:
Is this a debug message? Can this be removed?

Reply:
Oh yes! I'll remove it.